components/org.wso2.carbon.identity.oauth.ciba/src/main/java/org/wso2/carbon/identity/oauth/ciba/dao/CibaAuthMgtDAOImpl.java

```diff
@@ -0,0 +1,547 @@
+                                    "auth_req_id.");
+                        }
+                        return false;
+                    }
+                }
+
+                return false;
+            }
+        } catch (SQLException e) {
+
+            if (log.isDebugEnabled()) {
+                log.debug("Unsuccessful in checking whether provided hashedAuthReqId : " + hashedCibaAuthReqId +
+                        "exists.");
+            }
+            throw new CibaCoreException(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                    ErrorCodes.INTERNAL_SERVER_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * Returns CibaAuthCodeDOkey for the hash of CibaAuthReqId.
+     *
+     * @param hashedCibaAuthReqId hash of CibaAuthReqId.
+     * @return String Returns CibaAuthCodeDOKey.
+     * @throws CibaCoreException Exception thrown from CibaCore Component.
+     */
+    public String getCibaAuthCodeDOKey(String hashedCibaAuthReqId) throws CibaCoreException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.CibaSQLQueries.
+                    RETRIEVE_CIBA_AUTH_CODE_DO_KEY_BY_CIBA_AUTH_REQ_ID_HASH)) {
+                prepStmt.setString(1, hashedCibaAuthReqId);
+
+                ResultSet resultSet = prepStmt.executeQuery();
+
+                if (resultSet.next()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully returning CibaAuthCodeDOKey : " + resultSet.getString(1) + "for the " +
+                                "hashedCibaAuthReqId : " + hashedCibaAuthReqId);
+                    }
+                    return resultSet.getString(1);
+                } else {
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Could not find CibaAuthCodeDOKey for the hashedCibaAuthReqId : " +
+                                hashedCibaAuthReqId);
+                    }
+                    return null;
+                }
+            }
+        } catch (SQLException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error occured when finding CibaAuthCodeDOKey for the hashedCibaAuthReqId : " +
+                        hashedCibaAuthReqId);
+            }
+            throw new CibaCoreException(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+                    ErrorCodes.INTERNAL_SERVER_ERROR, e.getMessage());
+        }
+
+    }
+
+    /**
+     * Returns the lastPolledTime of tokenRequest with CibaAuthCodeDOKey.
+     *
+     * @param cibaAuthCodeDOKey Identifier of CibaAuthCodeDO.
+     * @return long Returns lastPolledTime.
+     * @throws CibaCoreException Exception thrown from CibaCore Component.
+     */
+    public long getCibaLastPolledTime(String cibaAuthCodeDOKey) throws CibaCoreException {
```

Comment:
IMO we do not need to include CIBA in the method name since the DAO name already suggests this is CIBA specific

Reply:
Refactored.